pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -772,5 +777,18 @@ func (r *Reconciler) addStoredAnnotations(ctx context.Context, o results.Object)
 		logger.Errorf("error patching object with stored annotation: %w ObjectName: %s", err, o.GetName())
 		return fmt.Errorf("error patching object with stored annotation: %w ObjectName: %s", err, o.GetName())
 	}
+
+	// Call AfterStorage callback if this is the first time we're marking it as stored after completion
+	// This ensures storage latency metrics are recorded exactly once per object when it transitions
+	// from "not stored after completion" to "stored after completion"
+	if stored.Value == "true" && r.AfterStorage != nil {
```

Comment:
I think we would never get here with the DisableStoringIncompleteRuns enabled. The first time addStoredAnnotations is invoked, it'll skip that block because the stored annotation won't be set. We'll never get back to this because the pipelinerun and taskrun reconcilers have this check `if pr.Annotations != nil && pr.Annotations[resultsannotation.Stored] == "true"` here https://github.com/tektoncd/results/blob/main/pkg/watcher/reconciler/pipelinerun/reconciler.go#L77 (the same for taskrun) and they'll skip the dynamic reconciler call all together.